DuplicateAssemblyScanner/DuplicateAssemblyScanner/Structs/ModAssembly.cs

```diff
@@ -0,0 +1,50 @@
+namespace DuplicateAssemblyScanner.Structs {
+    using DuplicateAssemblyScanner.Util;
+    using System.Reflection;
+    using static ColossalFramework.Plugins.PluginManager;
+
+    /// <summary>
+    /// Contains details of an assembly within a mod.
+    /// </summary>
+    public struct ModAssembly {
```

Comment:
Do not introduce `struct`s without a *real* need for that. Here, use a `class` instead.
Mutable structs are **evil**. Only use mutable structs for high-perf reasons when you need a continuous memory storage for a large array of data (obviously, this is not the case here).